src/transforms/aggregate.rs

```diff
@@ -92,22 +250,71 @@ impl Aggregate {
                     entry.insert((data, metadata));
                 }
             },
-            metric::MetricKind::Absolute => {
-                // Always replace/store
-                self.map.insert(series, (data, metadata));
-            }
-        };
-
-        emit!(AggregateEventRecorded);
+            metric::MetricKind::Absolute => {}
+        }
     }
 
     fn flush_into(&mut self, output: &mut Vec<Event>) {
         let map = std::mem::take(&mut self.map);
-        for (series, entry) in map.into_iter() {
-            let metric = metric::Metric::from_parts(series, entry.0, entry.1);
+        for (series, entry) in map.clone().into_iter() {
+            let mut metric = metric::Metric::from_parts(series, entry.0, entry.1);
+            if matches!(self.mode, AggregationMode::Diff) {
+                if let Some(prev_entry) = self.prev_map.get(metric.series()) {
+                    if metric.data().kind == prev_entry.0.kind {
+                        let _ = metric.subtract(&prev_entry.0);
```

Comment:
It is possible for the `subtract`. I have just added the same error that was used in previous implementation. I hope that is alright. Currently, I don't think it is possible for the `update` to return false, since `mean` and `stdev` are limited to just `Gauge` values, but I have added an error branch for them, just in case, if I have missed some case where we could get a failed update.